### PR TITLE
Add Event class and replace event string to constant of Event. Improve readability

### PR DESCRIPTION
**Pull request by huyanping**

### CODE REVIEW

File: src/CompositeStream.php [hunk 1]
```diff
@@ -18,10 +18,10 @@ public function __construct(ReadableStreamInterface $readable, WritableStreamInt
         Util::forwardEvents($this->readable, $this, array('data', 'end', 'error', 'close'));
         Util::forwardEvents($this->writable, $this, array('drain', 'error', 'close', 'pipe'));
 
-        $this->readable->on('close', array($this, 'close'));
-        $this->writable->on('close', array($this, 'close'));
+        $this->readable->on(Event::CLOSE, array($this, 'close'));
+        $this->writable->on(EVENT::CLOSE, array($this, 'close'));
```
Comment (steverhoades) [inline src/CompositeStream.php]: Typo

**Comment by huyanping** [comment]
Sorry about the typo, I had fix it.

**Comment by lysenkobv** [comment]
hi @huyanping 
Adding a class just to store consts is a **bad idea**.
We won't know what events exact object has. 

If event is emitted by WritableStream, it's required to add this event const to WritableStream. 
In this case we will know that this class emits it and we can use this event in inherited classes.

**Comment by huyanping** [comment]
hi @lysenkobv 

Thanks for your advise and it is a good idea. I will take time to do it.

**Comment by clue** [comment]
Thanks for your PR, but I'm having to vote :-1: on this for now, as I fail to see how this improves readability at all. Also, @lysenkobv gave some pointers as to how useful this would be anyway.

What do you think about this @huyanping, can this be closed?

**Pull request closed without merging**